File: dart/medium/findPopularChar.dart
class FindPopularChar {
  static String result(String s) {
    Map<String, int> map = {};
    int min = 0;
    String result = '';

    for (int i = 0; i < s.length; i++) {
      if (map.containsKey(s[i])) {
        map[s[i]] = map[s[i]]! + 1;
      } else {
        map[s[i]] = 1;
      }
    }

    for (MapEntry e in map.entries) {
      if (e.value > min) {
        min = e.value;
        result = e.key;
      }
    }

    return result;
  }
}

File: js/medium/findPopularChar.js
function findPopularChar(str) {
  var hashTable = {};
  var min = 0;
  var result = '';

  for (var i = 0; i < str.length; i++) {
    var char = str[i];
    hashTable[char] = hashTable[char] ? hashTable[char] + 1 : 1;
  }

  for (var key in hashTable) {
    if (hashTable[key] > min) {
      min = hashTable[key];
      result = key;
    }
  }

  return result;
}
console.log(f('ababa')); // 'a'
